lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -166,6 +166,17 @@ impl<'s> SegmentHolder {
             .chain(self.non_appendable_segments.iter())
     }
 
+    /// Iterate over all segments with their IDs
+    ///
+    /// Non-appendable first, then appendable.
+    pub fn iter_non_appendable_appendable(
```

Comment:
This is an ugly method. I want to find a better name for it before we actually merge.